gems/aws-sdk-core/spec/aws-sdk-core/credentials_provider_chain_spec.rb

```diff
@@ -0,0 +1,141 @@
+
+module AWS::SDK::Core
+  describe CredentialsProviderChain do
+    subject { described_class }
+
+    context '.call' do
+      file = File.join(File.dirname(__FILE__),
+                       'credentials-provider-chain-tests.json')
+      test_cases = JSON.load_file(file)
+
+      def credentials(values)
+        Identities::Credentials.new(
+          access_key_id: values['aws_access_key_id'],
+          secret_access_key: values['aws_secret_access_key'],
+          session_token: values['aws_session_token']
+        )
+      end
+
+      def mock_input(test_case, config)
+        test_case['input'].each do |input, values|
+          case input
+          when 'code'
+            config[:access_key_id] = values['aws_access_key_id']
+            config[:secret_access_key] = values['aws_secret_access_key']
+            config[:session_token] = values['aws_session_token']
+          when 'jvmSystemProperties'
+            # ignore
+          when 'env'
+            mock_env(values)
+          when 'configFiles'
+            mock_shared_config(values['aws'])
+          when 'stsResponse'
+            # assume_role_provider = double(
+            #   'AssumeRoleCredentialsProvider',
+            #   identity: credentials(values)
+            # )
+            # assume_role_provider_class = class_double(
+            #   'AssumeRoleCredentialsProvider',
+            #   from_profile: assume_role_provider
+            # )
+            # assume_role_web_identity_provider = double(
+            #   'AssumeRoleWebIdentityCredentialsProvider',
+            #   identity: credentials(values)
+            # )
+            # assume_role_web_identity_provider_class = class_double(
+            #   'AssumeRoleWebIdentityCredentialsProvider',
+            #   from_profile: assume_role_web_identity_provider,
+            #   from_env: assume_role_web_identity_provider
+            # )
+            # stub_const(
+            #   'AWS::SDK::STS::AssumeRoleCredentialsProvider',
+            #   assume_role_provider_class
+            # )
+            # stub_const(
+            #   'AWS::SDK::STS::AssumeRoleWebIdentityCredentialsProvider',
+            #   assume_role_web_identity_provider_class
+            # )
+            assume_role_provider = double(
+              'AssumeRoleCredentialsProvider',
+              identity: credentials(values)
+            )
+            assume_role_web_identity_provider = double(
+              'AssumeRoleWebIdentityCredentialsProvider',
+              identity: credentials(values)
+            )
+            # stub_const(
+            #   'AWS::SDK::STS::AssumeRoleCredentialsProvider',
+            #   assume_role_provider_class
+            # )
+            # stub_const(
+            #   'AWS::SDK::STS::AssumeRoleWebIdentityCredentialsProvider',
+            #   assume_role_web_identity_provider_class
+            # )
+            sts = AWS::SDK::STS
+            allow_any_instance_of(sts::AssumeRoleCredentialsProvider)
```

Comment:
I think I'd lean towards testing here with sts/sso/ssooidc loaded instead of trying to mock these.  I think its entirely reasonable to include these when running our tests and I think gives better test coverage of those providers + provider chain.

Reply:
I think the problem goes away if I just assert that from_profile is called